$NetBSD: patch-configure,v 1.14 2022/01/31 09:31:59 schmonz Exp $

- Builtin krb5-config in platforms such as solaris do not support
  the gssapi option, and need an explicit -lgss
- On Darwin, do not append custom CFLAGS.
  Also do not use DYLD_LIBRARY_PATH for ordinary configure checks.
- Do not strip debug flags.
- Support Minix.

--- configure.orig	2022-01-03 21:33:05.000000000 +0000
+++ configure
@@ -4243,6 +4243,7 @@ printf "%s\n" "$as_me: $xc_bad_var_msg l
         ;;
     esac
   done
+  xc_bad_var_cflags=no
   if test $xc_bad_var_cflags = yes; then
     { printf "%s\n" "$as_me:${as_lineno-$LINENO}: using CFLAGS: $CFLAGS" >&5
 printf "%s\n" "$as_me: using CFLAGS: $CFLAGS" >&6;}
@@ -8641,7 +8642,7 @@ else $as_nop
     lt_cv_sys_max_cmd_len=8192;
     ;;
 
-  bitrig* | darwin* | dragonfly* | freebsd* | netbsd* | openbsd*)
+  bitrig* | darwin* | dragonfly* | freebsd* | minix* | netbsd* | openbsd*)
     # This has been around since 386BSD, at least.  Likely further.
     if test -x /sbin/sysctl; then
       lt_cv_sys_max_cmd_len=`/sbin/sysctl -n kern.argmax`
@@ -9121,12 +9122,8 @@ linux* | k*bsd*-gnu | kopensolaris*-gnu 
   lt_cv_deplibs_check_method=pass_all
   ;;
 
-netbsd* | netbsdelf*-gnu)
-  if echo __ELF__ | $CC -E - | $GREP __ELF__ > /dev/null; then
-    lt_cv_deplibs_check_method='match_pattern /lib[^/]+(\.so\.[0-9]+\.[0-9]+|_pic\.a)$'
-  else
-    lt_cv_deplibs_check_method='match_pattern /lib[^/]+(\.so|_pic\.a)$'
-  fi
+netbsd* | netbsdelf*-gnu | minix*)
+  lt_cv_deplibs_check_method='match_pattern /lib[^/]+(\.so|_pic\.a)$'
   ;;
 
 newos6*)
@@ -13392,14 +13389,13 @@ _LT_EOF
       fi
       ;;
 
-    netbsd* | netbsdelf*-gnu)
-      if echo __ELF__ | $CC -E - | $GREP __ELF__ >/dev/null; then
-	archive_cmds='$LD -Bshareable $libobjs $deplibs $linker_flags -o $lib'
-	wlarc=
-      else
-	archive_cmds='$CC -shared $pic_flag $libobjs $deplibs $compiler_flags $wl-soname $wl$soname -o $lib'
-	archive_expsym_cmds='$CC -shared $pic_flag $libobjs $deplibs $compiler_flags $wl-soname $wl$soname $wl-retain-symbols-file $wl$export_symbols -o $lib'
-      fi
+    netbsd* | netbsdelf*-gnu | minix*)
+      archive_cmds='$CC -shared $pic_flag $libobjs $deplibs $compiler_flags ${wl}-soname $wl$soname -o $lib'
+      archive_expsym_cmds='$CC -shared $pic_flag $libobjs $deplibs $compiler_flags ${wl}-soname $wl$soname ${wl}-retain-symbols-file $wl$export_symbols -o $lib'
+      hardcode_libdir_flag_spec='${wl}-rpath,$libdir'
+      hardcode_direct=yes
+      hardcode_shlibpath_var=no
+      output_verbose_link_cmd=func_echo_all
       ;;
 
     solaris*)
@@ -14089,15 +14085,13 @@ printf "%s\n" "$lt_cv_irix_exported_symb
       esac
       ;;
 
-    netbsd* | netbsdelf*-gnu)
-      if echo __ELF__ | $CC -E - | $GREP __ELF__ >/dev/null; then
-	archive_cmds='$LD -Bshareable -o $lib $libobjs $deplibs $linker_flags'  # a.out
-      else
-	archive_cmds='$LD -shared -o $lib $libobjs $deplibs $linker_flags'      # ELF
-      fi
-      hardcode_libdir_flag_spec='-R$libdir'
+    netbsd* | netbsdelf*-gnu | minix*)
+      archive_cmds='$CC -shared $libobjs $deplibs $compiler_flags ${wl}-soname $wl$soname -o $lib'
+      archive_expsym_cmds='$CC -shared $libobjs $deplibs $compiler_flags ${wl}-soname $wl$soname ${wl}-retain-symbols-file $wl$export_symbols -o $lib'
+      hardcode_libdir_flag_spec='${wl}-rpath,$libdir'
       hardcode_direct=yes
       hardcode_shlibpath_var=no
+      output_verbose_link_cmd=func_echo_all
       ;;
 
     newsos6)
@@ -15207,6 +15201,18 @@ fi
   dynamic_linker='GNU/Linux ld.so'
   ;;
 
+minix*)
+  version_type=linux
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}$major ${libname}${shared_ext}'
+  soname_spec='${libname}${release}${shared_ext}$major'
+  dynamic_linker='Minix ld.elf_so'
+  shlibpath_var=LD_LIBRARY_PATH
+  shlibpath_overrides_runpath=yes
+  hardcode_into_libs=yes
+  ;;
+
 netbsdelf*-gnu)
   version_type=linux
   need_lib_prefix=no
@@ -17767,7 +17773,7 @@ squeeze() {
 
 
       #
-  if test "$compiler_id" != "unknown"; then
+  if false; then
     #
     tmp_save_CPPFLAGS="$CPPFLAGS"
     tmp_save_CFLAGS="$CFLAGS"
@@ -17938,7 +17944,7 @@ rm -f core conftest.err conftest.$ac_obj
     test "$tmp_compiler_works" = "yes"; then
 
    case $host_os in
-     darwin*) library_path_var=DYLD_LIBRARY_PATH ;;
+     darwin*) library_path_var=LD_LIBRARY_PATH ;;
      *)       library_path_var=LD_LIBRARY_PATH ;;
    esac
 
@@ -18029,13 +18035,6 @@ printf %s "checking if compiler accepts 
       tmp_options="$flags_dbg_yes"
     fi
     #
-    if test "$flags_prefer_cppflags" = "yes"; then
-      CPPFLAGS="$tmp_CPPFLAGS $tmp_options"
-      CFLAGS="$tmp_CFLAGS"
-    else
-      CPPFLAGS="$tmp_CPPFLAGS"
-      CFLAGS="$tmp_CFLAGS $tmp_options"
-    fi
     squeeze CPPFLAGS
     squeeze CFLAGS
   fi
@@ -18226,7 +18225,7 @@ rm -f core conftest.err conftest.$ac_obj
     test "$tmp_compiler_works" = "yes"; then
 
    case $host_os in
-     darwin*) library_path_var=DYLD_LIBRARY_PATH ;;
+     darwin*) library_path_var=LD_LIBRARY_PATH ;;
      *)       library_path_var=LD_LIBRARY_PATH ;;
    esac
 
@@ -19672,7 +19671,7 @@ rm -f core conftest.err conftest.$ac_obj
     test "$tmp_compiler_works" = "yes"; then
 
    case $host_os in
-     darwin*) library_path_var=DYLD_LIBRARY_PATH ;;
+     darwin*) library_path_var=LD_LIBRARY_PATH ;;
      *)       library_path_var=LD_LIBRARY_PATH ;;
    esac
 
@@ -20704,7 +20703,7 @@ printf "%s\n" "no" >&6; }
   tst_cflags="no"
   case $host_os in
     darwin*)
-      tst_cflags="yes"
+      tst_cflags="no"
       ;;
   esac
 
@@ -22521,7 +22520,7 @@ printf "%s\n" "$curl_cv_gclk_LIBS" >&6; 
 printf %s "checking if monotonic clock_gettime works... " >&6; }
 
    case $host_os in
-     darwin*) library_path_var=DYLD_LIBRARY_PATH ;;
+     darwin*) library_path_var=LD_LIBRARY_PATH ;;
      *)       library_path_var=LD_LIBRARY_PATH ;;
    esac
 
@@ -24231,7 +24230,7 @@ fi
 printf %s "checking if argv can be written to... " >&6; }
 
    case $host_os in
-     darwin*) library_path_var=DYLD_LIBRARY_PATH ;;
+     darwin*) library_path_var=LD_LIBRARY_PATH ;;
      *)       library_path_var=LD_LIBRARY_PATH ;;
    esac
 
@@ -24783,7 +24782,7 @@ printf "%s\n" "found" >&6; }
         LIBS="-lgss $LIBS"
         ;;
      *)
-        LIBS="-lgssapi $LIBS"
+        LIBS="-lgssapi -lkrb5 $LIBS"
         ;;
      esac
   fi
@@ -33995,7 +33994,7 @@ printf "%s\n" "#define HAVE_STRUCT_TIMEV
 printf %s "checking run-time libs availability... " >&6; }
 
    case $host_os in
-     darwin*) library_path_var=DYLD_LIBRARY_PATH ;;
+     darwin*) library_path_var=LD_LIBRARY_PATH ;;
      *)       library_path_var=LD_LIBRARY_PATH ;;
    esac
 
@@ -34459,7 +34458,7 @@ fi
 printf %s "checking if time_t is unsigned... " >&6; }
 
    case $host_os in
-     darwin*) library_path_var=DYLD_LIBRARY_PATH ;;
+     darwin*) library_path_var=LD_LIBRARY_PATH ;;
      *)       library_path_var=LD_LIBRARY_PATH ;;
    esac
 
@@ -37487,7 +37486,7 @@ rm -f core conftest.err conftest.$ac_obj
 printf %s "checking if getaddrinfo seems to work... " >&6; }
 
    case $host_os in
-     darwin*) library_path_var=DYLD_LIBRARY_PATH ;;
+     darwin*) library_path_var=LD_LIBRARY_PATH ;;
      *)       library_path_var=LD_LIBRARY_PATH ;;
    esac
 
@@ -38994,7 +38993,7 @@ rm -f core conftest.err conftest.$ac_obj
 printf %s "checking if getifaddrs seems to work... " >&6; }
 
    case $host_os in
-     darwin*) library_path_var=DYLD_LIBRARY_PATH ;;
+     darwin*) library_path_var=LD_LIBRARY_PATH ;;
      *)       library_path_var=LD_LIBRARY_PATH ;;
    esac
 
@@ -39239,7 +39238,7 @@ rm -f core conftest.err conftest.$ac_obj
 printf %s "checking if gmtime_r seems to work... " >&6; }
 
    case $host_os in
-     darwin*) library_path_var=DYLD_LIBRARY_PATH ;;
+     darwin*) library_path_var=LD_LIBRARY_PATH ;;
      *)       library_path_var=LD_LIBRARY_PATH ;;
    esac
 
@@ -39507,7 +39506,7 @@ rm -f core conftest.err conftest.$ac_obj
 printf %s "checking if inet_ntop seems to work... " >&6; }
 
    case $host_os in
-     darwin*) library_path_var=DYLD_LIBRARY_PATH ;;
+     darwin*) library_path_var=LD_LIBRARY_PATH ;;
      *)       library_path_var=LD_LIBRARY_PATH ;;
    esac
 
@@ -39767,7 +39766,7 @@ rm -f core conftest.err conftest.$ac_obj
 printf %s "checking if inet_pton seems to work... " >&6; }
 
    case $host_os in
-     darwin*) library_path_var=DYLD_LIBRARY_PATH ;;
+     darwin*) library_path_var=LD_LIBRARY_PATH ;;
      *)       library_path_var=LD_LIBRARY_PATH ;;
    esac
 
@@ -40813,7 +40812,7 @@ rm -f core conftest.err conftest.$ac_obj
 printf %s "checking if localtime_r seems to work... " >&6; }
 
    case $host_os in
-     darwin*) library_path_var=DYLD_LIBRARY_PATH ;;
+     darwin*) library_path_var=LD_LIBRARY_PATH ;;
      *)       library_path_var=LD_LIBRARY_PATH ;;
    esac
 
@@ -41218,7 +41217,7 @@ rm -f core conftest.err conftest.$ac_obj
 printf %s "checking if poll seems to work... " >&6; }
 
    case $host_os in
-     darwin*) library_path_var=DYLD_LIBRARY_PATH ;;
+     darwin*) library_path_var=LD_LIBRARY_PATH ;;
      *)       library_path_var=LD_LIBRARY_PATH ;;
    esac
 
@@ -43169,7 +43168,7 @@ printf "%s\n" "yes" >&6; }
 printf %s "checking if strerror_r seems to work... " >&6; }
 
    case $host_os in
-     darwin*) library_path_var=DYLD_LIBRARY_PATH ;;
+     darwin*) library_path_var=LD_LIBRARY_PATH ;;
      *)       library_path_var=LD_LIBRARY_PATH ;;
    esac
 
@@ -43285,7 +43284,7 @@ printf "%s\n" "yes" >&6; }
 printf %s "checking if strerror_r seems to work... " >&6; }
 
    case $host_os in
-     darwin*) library_path_var=DYLD_LIBRARY_PATH ;;
+     darwin*) library_path_var=LD_LIBRARY_PATH ;;
      *)       library_path_var=LD_LIBRARY_PATH ;;
    esac
 
